sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_helpers.cs

```diff
@@ -354,5 +373,63 @@ private string GenerateTokenFromAzureKeyCredential(DateTimeOffset expiresAt, str
 
             return jwt.BuildString();
         }
+
+        private string GetRelativeClientEndpoint(ClientType endpointType) => endpointType switch
+        {
+            ClientType.Default => $"client/hubs/{_hub}",
+            ClientType.MQTT => $"clients/mqtt/hubs/{_hub}",
+            _ => throw new ArgumentOutOfRangeException(nameof(endpointType))
+        };
+
+        private async Task<string> GetClientAccessTokenCore(
+             DateTimeOffset expiresAt,
+             string userId,
+             IEnumerable<string> roles,
+             IEnumerable<string> groups,
+             ClientType clientType,
+             bool async,
+             CancellationToken cancellationToken = default)
+        {
+            if (clientType == ClientType.MQTT && _apiVersionEnum < WebPubSubServiceClientOptions.ServiceVersion.V2024_01_01)
+            {
+                throw new NotSupportedException($"Getting client access token for MQTT is supported since API version {WebPubSubServiceClientOptions.ServiceVersion.V2024_01_01.ToVersionString()}. Current version is {_apiVersion}.");
+            }
```

Comment:
Will the service validate this for you and return a clear error?  If so, we should let the service do as much validation as possible.  Client-side validation typically gets out of date, takes effort to maintain, and can cause compat issues over time.  See https://azure.github.io/azure-sdk/general_implementation.html and https://azure.github.io/azure-sdk/dotnet_introduction.html#parameter-validation.

Reply:
Currently our service doesn't reject the request if it encounters unrecognized parameters, therefore, if the `clientType` parameter is applied to the old api version, no error is thrown. I'm just afraid that we start to block requests with unrecogized parameters in the old api versions, it would be a breaking change.

Reply:
That's unfortunate.  You definitely can't change the service behavior of existing versions and even validating now when you didn't in the past would break anyone upgrading.  I think you're stuck with that behavior so the client-side validation might be a necessary evil here.